src/Resources/contao/widgets/PickerWidget.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+/**
+ * Contao Open Source CMS
+ *
+ * Copyright (c) 2005-2017 Leo Feyer
+ *
+ * @license LGPL-3.0+
+ */
+
+namespace Contao;
+
+
+/**
+ * Provide methods to handle input field "picker".
+ *
+ * @property boolean        $multiple
+ * @property array          $rootNodes
+ * @property string         $fieldType
+ * @property string         $foreignTable
+ * @property string         $context
+ * @property string         $labelField
+ * @property string         $icon
+ * @property boolean        $sortable
+ * @property array|callable $label_callback
+ *
+ * @author Leo Feyer <https://github.com/leofeyer>
+ * @author Kamil Kuzminski <https://github.com/qzminski>
+ */
+class PickerWidget extends \Widget
+{
+
+	/**
+	 * Submit user input
+	 * @var boolean
+	 */
+	protected $blnSubmitInput = true;
+
+	/**
+	 * Template
+	 * @var string
+	 */
+	protected $strTemplate = 'be_widget';
+
+
+	/**
+	 * Load the database object
+	 *
+	 * @param array $arrAttributes
+	 */
+	public function __construct($arrAttributes=null)
+	{
+		$this->import('Database');
+		parent::__construct($arrAttributes);
+	}
+
+
+	/**
+	 * Return an array if the "multiple" attribute is set
+	 *
+	 * @param mixed $varInput
+	 *
+	 * @return mixed
+	 */
+	protected function validator($varInput)
+	{
+		$this->checkValue($varInput);
+
+		if ($this->hasErrors())
+		{
+			return '';
+		}
+
+		// Return the value as usual
+		if ($varInput == '')
+		{
+			if ($this->mandatory)
+			{
+				$this->addError(sprintf($GLOBALS['TL_LANG']['ERR']['mandatory'], $this->strLabel));
+			}
+
+			return '';
+		}
+		elseif (strpos($varInput, ',') === false)
+		{
+			return $this->multiple ? array(intval($varInput)) : intval($varInput);
+		}
+		else
+		{
+			$arrValue = array_map('intval', array_filter(explode(',', $varInput)));
+
+			return $this->multiple ? $arrValue : $arrValue[0];
+		}
+	}
+
+
+	/**
+	 * Check the selected value
+	 *
+	 * @param mixed $varInput
+	 */
+	protected function checkValue($varInput)
```

Comment:
I think this should be `validateValue($varInput)`

Reply:
I have copied it "as is" from the page tree widget. I can change the method name here but then it will be a mismatch between those two widgets. Should not be a problem but if 90% is already the same...

Reply:
Then we should keep it, sorry I didn't see that.